Normative: add Math.sumPrecise

bakkot opened the pull request:
Pending [stage 4](https://github.com/tc39/proposal-math-sum/issues/19).

michaelficarra commented on spec.html:
```diff
@@ -33213,6 +33213,50 @@ <h1>Math.sinh ( _x_ )</h1>
+            1. Set _next_ to ? IteratorStepValue(_iteratorRecord_).
+            1. If _next_ is not ~done~, then
+              1. Set _count_ to _count_ + 1.
+              1. If _count_ ≥ 2<sup>53</sup>, then
+                1. Let _error_ be ThrowCompletion(a newly created *RangeError* object).
+                1. Return ? IteratorClose(_iteratorRecord_, _error_).
+              1. NOTE: The above case is not expected to be reached in practice and is included only so that implementations may rely on inputs being "reasonably sized" without violating this specification.
```
```suggestion
              1. If _count_ ≥ 2<sup>53</sup>, then
                1. NOTE: This step is not expected to be reached in practice and is included only so that implementations may rely on inputs being "reasonably sized" without violating this specification.
                1. Let _error_ be ThrowCompletion(a newly created *RangeError* object).
                1. Return ? IteratorClose(_iteratorRecord_, _error_).
```

linusg commented on spec.html:
```diff
@@ -33213,6 +33213,50 @@ <h1>Math.sinh ( _x_ )</h1>
         </emu-note>
       </emu-clause>
 
+      <emu-clause id="sec-math.sumprecise">
```
After Math.sqrt for correct ordering

bakkot commented:
Addressed comments. I also pushed up [a small tweak](https://github.com/tc39/ecma262/pull/3654/commits/51d7a5920410de6491972d7f180d0e461833741e) to how `_count_` is handled to match the prevailing style as seen in e.g. the [GroupBy AO](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-groupby).
